static/js/utils/string_utils.ts

```diff
@@ -16,12 +16,30 @@
 
 import _ from "lodash";
 
+/**
+ * If the date is in YYYY-MM format, returns YYYY-MMM format.
+ * This is to improve readability of dates in chart titles.
+ */
+export function formatDate(strDate: string): string {
+  const regexDate = /^\d{4}-\d{1,2}$/;
+  if (!regexDate.test(strDate)) {
+    return strDate;
+  } else {
+    const dt = new Date(strDate);
+    return (
+      new Intl.DateTimeFormat("en-US", { year: "numeric" }).format(dt) +
+      "-" +
+      new Intl.DateTimeFormat("en-US", { month: "short" }).format(dt)
```

Comment:
I think this format also looks unnatural. How about using the formatter to generate the entire final string:

```
new Intl.DateTimeFormat("en-US", { year: "numeric", month: "short"}).format(dt)
```

Reply:
We tried your suggestion of using the formatter once. But the resulting date appears in the format "MMM YYYY" (with a space in between). We couldn't find any options to force our format of YYYY-MMM.

Also, added a test for this function in utils.

Reply:
Yes, the dash there looks unusual as well, when paired with the spelled out date. We can try this for now and adjust as we go.